<!-- jplaced@unizar.es
 2022, Universidad de Zaragoza -->

<!--  Launch file for single robot on Gazebo -->
<launch>
  <env name="GAZEBO_RESOURCE_PATH" value="$(find aslam_rosbot)/launch/include/meshes"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="gui" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="debug" value="0"/>
    <arg name="world_name" value="$(find aslam_rosbot)/launch/include/worlds/willowgarage_modified.world"/>
    <arg name="verbose" value="true"/>
  </include>

  <!-- <group ns="/robot_1"> -->
  <include file="$(find aslam_rosbot)/launch/include/robot.launch.xml">
    <arg name="robot_name" value="robot_1"/>
    <arg name="init_pose" value="-x 0.0 -y 0.0 -z 0.0 -R 0.0 -P 0.0 -Y 0.0"/>
  </include>

  <node pkg="slam_karto_g2o" type="slam_karto_g2o" name="slam_karto_g2o" output="screen">
    <remap from="scan" to="base_scan"/>
    <rosparam command="load" file="$(find aslam_rosbot)/param/mapping_karto_g2o.yaml" />
    <param name="odom_frame" value="odom"/>
    <param name="map_frame" value="map"/>
    <param name="base_frame" value="base_link"/>
  </node>


  <node name="obstacle_avoidance_node" pkg="aslam_rosbot" type="obsavoidence.py" output="screen" respawn="true"> </node>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find aslam_rosbot)/launch/include/rviz_config/single.rviz">
    <remap from="move_base_simple/goal" to="move_base_simple/goal"/>
  </node>

</launch>
